app/models/environment_class.rb

```diff
@@ -28,4 +29,10 @@ def self.is_in_any_environment(puppetclass, puppetclass_lookup_key)
   def self.key_in_environment(env, puppetclass, puppetclass_lookup_key)
     EnvironmentClass.where(:environment_id => env, :puppetclass_id => puppetclass, :puppetclass_lookup_key_id => puppetclass_lookup_key).first
   end
+
+  def delete_orphaned_lookup_keys
+    if puppetclass_lookup_key.environment_classes.empty?
```

Comment:
What's the difference between this and `belongs_to :puppetclass_lookup_key, :dependent => :destroy` ?